temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/ChooseIconModal.kt

```diff
@@ -222,7 +224,7 @@ private fun LazyListScope.addIconsRowIfNotEmpty(
     if (rowAcc.isNotEmpty()) {
         item {
             IconsRow(
-                icons = rowAcc,
+                icons = rowAcc.toImmutableList(),
```

Comment:
@AMHaroun don't do it like this - `.toImmutabpeList()` is a heavy computations because it recreates the entire list.

There do two solutions:
1. Do this conversion in the ViewModel (non composable)
2. Use `remember(rowAcc) { ... }` to compute it only once (this can still be an issue because if `rowAcc` is unstable this will always be re-computed)

Reply:
Here moving to the ViewModel can be hard cuz our legacy code is garbage 😀

Reply:
I tried to implement the second solution because it was simpler.